frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -48,22 +53,95 @@ export const surveysLogic = kea<surveysLogicType>([
+                surveysCount: 0,
+                searchSurveys: [] as Survey[],
+                searchSurveysCount: 0,
+            },
             loadSurveys: async () => {
-                const responseSurveys = await api.surveys.list()
-                return responseSurveys.results
+                const response = await api.surveys.list()
+                return {
+                    ...values.data,
+                    surveys: response.results,
+                    surveysCount: response.count,
+                }
+            },
+            loadNextPage: async () => {
+                const offset = values.data.surveys.length
+                const response = await api.surveys.list({
+                    limit: SURVEY_PAGE_SIZE,
+                    offset,
+                })
+
+                return {
+                    ...values.data,
+                    surveys: [...values.data.surveys, ...response.results],
+                    surveysCount: response.count,
+                }
+            },
+            loadSearchResults: async () => {
+                const trimmedSearchTerm = values.searchTerm?.trim() || ''
+                if (trimmedSearchTerm === '') {
+                    return {
+                        ...values.data,
+                        searchSurveys: [],
+                        searchSurveysCount: 0,
+                    }
+                }
+
+                // Only do backend search if we have more total items than the page size
+                if (values.data.surveysCount <= SURVEY_PAGE_SIZE) {
+                    return values.data
+                }
+
+                const response = await api.surveys.list({
+                    limit: SURVEY_PAGE_SIZE,
+                    search: trimmedSearchTerm,
+                })
+
+                return {
+                    ...values.data,
+                    searchSurveys: response?.results || [],
+                    searchSurveysCount: response?.count || 0,
+                }
+            },
+            loadNextSearchPage: async () => {
+                const offset = values.data.searchSurveys.length
+                const response = await api.surveys.list({
+                    search: values.searchTerm,
+                    limit: SURVEY_PAGE_SIZE,
+                    offset,
+                })
+
+                return {
+                    ...values.data,
+                    searchSurveys: [...values.data.searchSurveys, ...response.results],
+                    searchSurveysCount: response.count,
+                }
             },
             deleteSurvey: async (id) => {
                 await api.surveys.delete(id)
-                return values.surveys.filter((survey) => survey.id !== id)
+                return {
+                    ...values.data,
+                    surveys: values.data.surveys.filter((survey) => survey.id !== id),
+                    searchSurveys: values.data.searchSurveys.filter((survey) => survey.id !== id),
```

Comment:
this and `updateSurvey` are repeating, we can just have a method passing the list to be filtered out